src/tools/namespace_tool_core.cpp

```diff
@@ -125,30 +125,54 @@ int NameSpaceToolCore::GetFileSegments(const std::string& fileName,
     return 0;
 }
 
-int NameSpaceToolCore::CleanRecycleBin(const std::string& dirName) {
+int NameSpaceToolCore::CleanRecycleBin(const std::string& dirName,
+                                       const uint64_t expireTime) {
     std::vector<FileInfo> files;
     int res = client_->ListDir(curve::mds::RECYCLEBINDIR, &files);
     if (res != 0) {
         std::cout << "List RecycleBin fail!" << std::endl;
         return -1;
     }
+
+    auto needDelete = [](const FileInfo &fileInfo,
+                         uint64_t now,
+                         uint64_t expireTime) -> bool {
+        auto filename = fileInfo.filename();
+        std::vector<std::string> items;
+        ::curve::common::SplitString(filename, "-", &items);
+
+        uint64_t dtime;
+        auto n = items.size();
+        auto id = std::to_string(fileInfo.id());
+        if (n >= 2 && items[n - 2] == id
+            && ::curve::common::StringToUll(items[n - 1], &dtime)
+            && now - dtime < expireTime) {
+            return false;
+        }
+
+        return true;
+    };
+
     bool success = true;
+    uint64_t now = ::curve::common::TimeUtility::GetTimeofDaySec();
+    // Ensure the dirname end with on slash
+    auto dirname = dirName + "/";
+    dirname = dirname.substr(0, dirname.find_last_not_of('/') + 2);
```

Comment:
if the dirname is uncorrect return error

Reply:
> if the dirname is uncorrect return error

I will deal with it more elegant.